code/game/turfs/turf.dm

```diff
@@ -167,14 +167,16 @@ GLOBAL_LIST_EMPTY(station_turfs)
 	SEND_SIGNAL(src, COMSIG_TURF_MULTIZ_NEW, T, dir)
 
 ///returns if the turf has something dense inside it. if exclude_mobs is true, skips dense mobs like fat yoshi. if exclude_object is true, it will exclude the excluded_object you sent through
-/turf/proc/is_blocked_turf(exclude_mobs, excluded_object)
+/turf/proc/is_blocked_turf(exclude_mobs, list/excluded_objects = list())
 	if(density)
 		return TRUE
 	for(var/i in contents)
 		var/atom/thing = i
-		if(thing == excluded_object)
-			continue
-		if(thing.density && (!exclude_mobs || !ismob(thing)))
+		var/excuded = FALSE
+		for(var/excluded in excluded_objects)
+			if(istype(thing, excluded))
```

Comment:
```suggestion
			if(ispath(thing, excluded))
```
If you are using type paths, `ispath()` needs to be used, instead of `istype()`

Reply:
It is done, thanks i now understand what and why here.